### PR TITLE
fix/#979 fix springboot running plain jar instead of executable jar

### PR DESCRIPTION
**Pull request by fattila16**
This PR solves the following issue: https://github.com/railwayapp/nixpacks/issues/979

~~**Disclaimer**:  This drops support of any spring version that does not have a `bootJar` task, mainly spring v1~~

* ~~Adjust the `JavaProvider` so that it uses `bootJar` to build an executable JAR instead of `build` which generates a `*plain.jar` also when spring and gradle is detected~~
* ~~Add e2e test cases for spring v2 and v3, also remove spring v1~~
* ~~Update the docs of the `JavaProvider` so that it has information regarding the spring version support, and the build commands~~

As per the comments below PR implements a different solution now:

* Adjust the `JavaProvider` so that if gradle and spring is detected then the start cmd includes a one line to find the executable jar and blacklists the plain.jar
* Add e2e test cases for spring v1, v2 and v3
* Update the docs of the `JavaProvider` so that it includes description for the new start cmd if gradle and spring is detected

- [x] Unit tests are added
- [x] E2E tests are added
- [x] Docs are updated

### CODE REVIEW

File: src/providers/java.rs [hunk 1]
```diff
@@ -95,11 +96,20 @@ impl JavaProvider {
         }
     }
 
+    fn get_gradle_build_cmd(&self, build_gradle_content: &String, gradle_exe: &String) -> String {
+        if self.is_using_spring_boot(build_gradle_content) {
```
Comment (coffee-cup) [inline src/providers/java.rs:100]: Is it possible to detect if Spring v1 is used and only use `bootJar` if it is version 2+?

Comment (fattila16) [reply src/providers/java.rs:100]: One can read it from the `build.gradle` file, however I would not go that way. There can be edge cases, where a gradle plugin update or a spring version update changes the versioning scheme. Also one can define it in kotlin and groovy. That makes capturing this a little bit finnicky and I would not want to rely on a bunch of regexp expressions.

If `nixpacks` would like to support spring v1 as well, there is an alternative solution to what I've provided in this PR. Which is the following:

* The problem is that a `*plain.jar` is being generated along with an executable `.jar`
* The start command can point to the `*plain.jar` instead of the executable
* As a solution one can adjust the start command with the following one-liner: 
`java $JAVA_OPTS -jar -Dserver.port=$PORT $(ls -1 build/libs/*jar | grep -v plain)`
* So basically instead of configuring gradle to create only an executable jar, we use `ls` and `grep` to find the executable `jar` instead of the `plain` one

What do you think about this approach?

Comment (coffee-cup) [reply src/providers/java.rs:100]: I think that solution works! And yeah, supporting v1 isn't 100% necessary, but backwards compatibility is always nice, especially if the solution is as simple as greping for the executable jar

Comment (fattila16) [reply src/providers/java.rs]: I've updated the PR with the new solution

Comment (fattila16) [reply src/providers/java.rs]: Is it possible for me to get a review here? And then I can check whether all the pipelines are green or not. Thank you!

**Review by coffee-cup — APPROVED** [review]

**Comment by coffee-cup** [comment]
Thank you 🎉
